src/app_service/service/stickers/service.nim

```diff
@@ -0,0 +1,237 @@
+import NimQml, Tables, json, sequtils, strformat, chronicles, strutils, atomics, sets, strutils, tables
+import status/types/[rpc_response, sticker, pending_transaction_type]
+
+import eventemitter
+import ../../tasks/[qt, threadpool]
+
+import status/stickers as status_stickers_legacy
+import status/statusgo_backend_new/stickers as status_stickers
+import status/statusgo_backend/chat as status_chat
+import status/chat as status_chat_old
+import ./dto/stickers
+import ../settings/service as settings_service
+import ../wallet_account/service as wallet_account_service
+import ../transaction/service as transaction_service
+import ../chat/service as chat_service
+import status/utils as status_utils
+
+export StickerDto
+export StickerPackDto
+
+include async_tasks
+
+logScope:
+  topics = "stickers-service"
+
+type
+  StickerPackLoadedArgs* = ref object of Args
+    stickerPack*: StickerPackDto
+    isInstalled*: bool
+    isBought*: bool
+    isPending*: bool
+  StickerGasEstimatedArgs* = ref object of Args
+    estimate*: int
+    uuid*: string
+
+# Signals which may be emitted by this service:
+const SIGNAL_STICKER_PACK_LOADED* = "SIGNAL_STICKER_PACK_LOADED"
+const SIGNAL_ALL_STICKER_PACKS_LOADED* = "SIGNAL_ALL_STICKER_PACKS_LOADED"
+const SIGNAL_STICKER_GAS_ESTIMATED* = "SIGNAL_STICKER_GAS_ESTIMATED"
+
+QtObject:
+  type Service* = ref object of QObject
+    threadpool: ThreadPool
+    installedStickerPacks*: Table[int, StickerPackDto]
+    availableStickerPacks*: Table[int, StickerPackDto]
+    purchasedStickerPacks*: seq[int]
+    recentStickers*: seq[StickerDto]
+    events: EventEmitter
+    settingsService: settings_service.Service
+    walletAccountService: wallet_account_service.Service
+    transactionService: transaction_service.Service
+    chatService: chat_service.Service
+
+  proc delete*(self: Service) =
+    self.QObject.delete
+
+  proc newService*(
+      events: EventEmitter,
+      threadpool: ThreadPool,
+      settingsService: settings_service.Service,
+      walletAccountService: wallet_account_service.Service,
+      transactionService: transaction_service.Service,
+      chatService: chat_service.Service
+      ): Service =
+    new(result, delete)
+    result.QObject.setup
+    result.events = events
+    result.threadpool = threadpool
+    result.settingsService = settingsService
+    result.walletAccountService = walletAccountService
+    result.transactionService = transactionService
+    result.chatService = chatService
+    result.availableStickerPacks = initTable[int, StickerPackDto]()
+    result.installedStickerPacks = initTable[int, StickerPackDto]()
+    result.purchasedStickerPacks = @[]
+    result.recentStickers = @[]
+
+  proc init*(self: Service) =
+    discard
+  
+  proc getInstalledStickerPacks*(self: Service): Table[int, StickerPackDto] =
```

Comment:
@jrainville since the sticker pack is available from the very first start, the better place to initiate it on the start is `init` proc here (call should be done from the AppController but only if stickers are enabled) and later we need just to maintain that table properly here in this class and notify interested parts about the change.

Reply:
I moved the `obtainAvailableStickerPacks` call to the init above. Let me know if that's what you meant.